h5pyd/_hl/attrs.py

```diff
@@ -166,24 +166,90 @@ def __getitem__(self, name):
 
         return arr
 
+    def get_attributes(self, names=None, pattern=None, limit=None, marker=None, use_cache=True):
+        """
+        Get all attributes or a subset of attributes from the target object.
+        If 'use_cache' is True, use the objdb cache if available.
+        The cache cannot be used with pattern, limit, or marker parameters.
+        - if 'pattern' is provided, retrieve all attributes with names that match the pattern
+          according to Unix pathname pattern expansion rules.
+        - if 'limit' is provided, retrieve at most 'limit' attributes.
+        - if 'marker' is provided, retrieve attributes whose names occur after the name 'marker' in the target object
+        """
+        if use_cache and (pattern or limit or marker):
+            raise ValueError("use_cache cannot be used with pattern, limit, or marker parameters")
+
```

Comment:
It was initially used to stop the cache from being used with requests that have parameters which affect the result - this is now done in the HttpConn. It can be removed now